src/chrome/content/code/HTTPS.js

```diff
@@ -39,7 +39,53 @@ const HTTPS = {
    *   untouched or aborted.
    */
   replaceChannel: function(applicable_list, channel, httpNowhereEnabled) {
-    var blob = HTTPSRules.rewrittenURI(applicable_list, channel.URI.clone());
+    var that = this;
+    // If the callback gets called immediately (not async), it will be called
+    // before the return from the function sets this variable, so we default it
+    // to true.
+    var callbackedImmediate = true;
+    callbackedImmediate = HTTPSRules.rewrittenURI(applicable_list, channel.URI.clone(), function(blob) {
```

Comment:
nit: with complex async logic like this, I'd find it more legible if the callback was assigned to a variable with a good name rather than being defined in the middle of the `rewrittenURI` call.